### ISSUE TITLE
Load file partially to avoid buffer overload

### ISSUE THREAD

**Issue by kldjonge** [lbl-srg/BuildingsPy#435]
Is it possible to only load the .mat file partially? 
When only wanting to extract a few variables it seems that still the whole .mat file is loaded.  This results in a buffer overload error when the .mat file gets to big (e.g. when also the protected variables were stored for debugging purposes) which might be avoided when only the necessary data is loaded.

I got the inspiration from the optional 'variable_names' argument in de scipy.io.loadmat class.
https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.loadmat.html

**Comment by mwetter** [comment]
Did you check if DyMat supports this? Reading the `.mat` files is done through the `DyMat` library.

**Comment by kldjonge** [comment]
I checked indeed and it doesn't i'm afraid.  DyMat uses the scipy.io.loadmat  I mentioned but doesn't use the 'variables _names' option so I think it always loads the whole file.

**Comment by Walshest** [comment]
Hi @kldjonge , 
Did you find a solution to the buffer error? I have the same issue with larger files: https://github.com/lbl-srg/BuildingsPy/issues/474

**Comment by kldjonge** [comment]
@Walshest, I haven't. 
DyMat (and later BuildingsPy) should be updated so that the optional argument 'variables_names' of scipy.io.loadmat could be used.

**Comment by thorade** [comment]
https://stackoverflow.com/questions/70513151/opening-big-result-mat-file-raises-buffer-is-too-small-for-requested-array
https://github.com/jraedler/DyMat/issues/4

**Comment by kldjonge** [comment]
@thorade, indeed, exporting a smaller .MAT file is what I ended up doing. Well, I first simulated and stored all variables (might need them for later analysis or used to check the internal workings of the model/controls) then I plotted the once I needed and exported a .MAT file with only those variables.

This wasn't convient though and is a lot of manual work for bath simulations. It would be nice to be able to load only a subset of the original .MAT file directly in Python. I haven't actually tried this, but as scipy.io.loadmat can do it, probably the fix is not that far away. Although I don't think there is any active developement for DyMat.

**Comment by bravache** [comment]
Thanks @Walshest , I agree that it would probably be better to allow BuildingsPy to handle larger results. I will take a look at it.
